src/main/java/mate/academy/intro/service/BookServiceImpl.java

```diff
@@ -36,4 +43,31 @@ public BookDto getById(Long id) {
         );
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public BookDto updateById(Long id, UpdateBookRequestDto updatedBookDataDto) {
+        Book existingBook = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book by id: " + id));
+        Book updateBookData = bookMapper.toModel(updatedBookDataDto);
+        BeanUtils.copyProperties(updateBookData, existingBook, getFieldsWithNull(updateBookData));
```

Comment:
create in BookMapper
`void updateBookFromDto(CreateBookRequestDto book, @MappingTarget Book entity);`

```suggestion
        bookMapper.updateBookFromDto(updatedBookDataDto, book);
```